### PR TITLE
Will/mobile-segment-events

### PR DESCRIPTION
**Pull request by camerow**
adds some basic event tracking for mobile

### CODE REVIEW

File: apps/mobile/src/app/_layout.tsx [hunk 1]
```diff
@@ -41,6 +43,15 @@ initAppServices();
...
 
 export default function RootLayout() {
+  const pathname = usePathname();
+  const params = useGlobalSearchParams();
+
+  useEffect(() => {
+    analytics.screen(pathname, {
+      params,
+    });
+  }, [pathname, params]);
```
Comment (kyranjamie) [inline apps/mobile/src/app/_layout.tsx:53]: Even though a few LOC, this'd be great as its own hook with api like

`useOnRouteChange(route => analytics.screen(pathname, { params }))`

File: packages/analytics/src/events.ts [hunk 2]
```diff
@@ -102,3 +106,28 @@ interface HistoricalEvents {
...
+  | { fiat_currency: string }
+  | {
+      network:
+        | 'mainnet'
+        | 'testnet'
+        | 'testnet4'
+        | 'signet'
+        | 'sbtcTestnet'
+        | 'sbtcDevenv'
+        | 'devnet';
```
Comment (kyranjamie) [inline packages/analytics/src/events.ts:124]: We could pull values from models such that this'll update when we add new networks

Comment (camerow) [reply packages/analytics/src/events.ts:124]: Done

**Comment by pete-watters** [comment]
@camerow just re-visiting this as I think we aren't respecting the users preferences here, is that correct? 

We have a setting for it `analyticsPreference` but it seems like we don't use it here and always track? 

Screenshot 2024-12-09 at 10 56 55

**Pull request merged**